### PR TITLE
added slider for zooming on graph view

### LINKED ISSUE #482: On dag display include zoom control
The dag display would be easier to use with an explicit zoom control instead of just pinching gestures.

### PR DESCRIPTION
**Pull request by KevYuen**
This closes #482.

There is now a slider that allow users to zoom in and out of the DAG graph. This is beneficial to user that cannot easily scroll(laptop with bad trackpad, tablet, etc...). 

Sorry, linter may have also added some :gift:

### CODE REVIEW

File: airflow/www/templates/airflow/graph.html [hunk 1]
```diff
@@ -258,5 +270,22 @@
...
     }
 
+    function zoomWithSlider() {
+        var svg = d3.select("body").select("svg");
+        var container = svg.select("g");
+        var scale = 0.5 + parseInt(slider.getValue())/10;
+
+        container.attr("transform", "translate(20, 20) scale(" + scale + ")");
+
+        // var h = svg.attr("height"), w = parseInt(svg.style("width"));
```
Comment (KevYuen) [inline airflow/www/templates/airflow/graph.html]: The commented out code here is for zooming with the centre of the svg as a focus point. In testing, this seems to be less useful then simply zooming at "20, 20" as things tends to fly off screen.

Would be interested in your input @mistercrunch 

Let me know what you'd prefer and I can make the changes.

**Comment by mistercrunch** [comment]
Looks pretty good but not 100% intuitive. I think ideally you'd compute the center of the graph and have that be the point of focus, but I'm not sure how hard it would be to do...

**Comment by KevYuen** [comment]
@mistercrunch: The commented out code I had inside zoomWithSlider(), zooms with the center of the svg-container as the point of focus. Not sure how hard it would be to zoom from the center of the graph.

**Comment by mistercrunch** [comment]
I also noticed that this PR changes the current behavior of the mousewheel zoom, I prefer the previous behavior as far as the mousewheel goes.

**Comment by KevYuen** [comment]
I will hunt down a mouse and test this out. I don't own a mouse haha!

**Comment by mistercrunch** [comment]
Wait how do you move the slider that you added without a mouse?!

**Comment by KevYuen** [comment]
trackpad! I didn't find any difference in the zoom behaviour with a trackpad. So I figure it's mouse specific scrolling thing. 

Hmmm is the difference that there are now 2 zoom boxes? Ex. You can slide the slider to 10(max value), then pinch on the track pad or scroll with a mouse to continue zooming in?

Also, if by "center of the graph" you mean center of the svg-container, I can do that.

**Comment by mistercrunch** [comment]
The difference to me is that the zooming with the trackpad or mousewheel would be done relative to the position of the mouse pointer before, now it's based on a fixed point on the upper right corner (from memory).

By center of the graph I mean center of the graph itself, not the viewport, so that if the center of the graph is anywhere in the viewport, any level of zooming will keep that middle point in the same spot in the viewport.

**Comment by KevYuen** [comment]
I will have to look into how to do that. Will commit the code once I figure it out. Probably over the weekend.

**Comment by KevYuen** [comment]
@mistercrunch The trackpad or mousewheel is relative to the position of the mouse on my end. 

I couldn't figure out how to make the zoom based on the centre of the graph itself due to changes in the position that can be made based on panning.

**Pull request closed without merging**